node/cn/api_tracer.go

```diff
@@ -802,22 +802,31 @@ func (api *PrivateDebugAPI) traceTx(ctx context.Context, message blockchain.Mess
 func (api *PrivateDebugAPI) stateAt(block *types.Block, reexec uint64) (*state.StateDB, func(), error) {
 	var stateDB *state.StateDB
 
-	// If we have the state fully available, use that.
+	// If we have the state fully available in cachedNode, use that.
 	stateDB, err := api.cn.blockchain.StateAtWithGCLock(block.Root())
-	if err != nil {
-		emptyFn := func() {}
+	if err == nil {
+		logger.Debug("Get stateDB from stateCache", "block", block.NumberU64())
+		// During this processing, this lock will prevent to evict the state.
+		return stateDB, stateDB.UnlockGCCachedNode, nil
+	}
 
-		// If no state is locally available, the desired state will be generated.
-		stateDB, err = api.computeStateDB(block, reexec)
-		if err != nil {
-			return nil, emptyFn, err
-		}
-		logger.Debug("Get stateDB by computeStateDB", "block", block.NumberU64())
+	emptyFn := func() {}
```

Comment:
How about declaring this as a static variable?

Reply:
What do you mean a static variable?
It means to declare a global variable? :)

Reply:
Yes, a package variable.

Reply:
It is only used in this function. So I think we don't need to declare it as a package variable. :)